src/features/filter.ts

```diff
@@ -1,10 +1,50 @@
-export const actions = { /* put action creators here */};
+import { FilterType } from '../types/FilterType';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetQueryAction = { type: 'filter/SET_QUERY', payload: string };
+type SetStatusAction = { type: 'filter/SET_STATUS', payload: FilterType };
+type SetClearAction = { type: 'filter/SET_CLEAR' };
+
+const setQuery = (query: string): SetQueryAction => ({
+  type: 'filter/SET_QUERY',
+  payload: query,
+});
+const setStatus = (status: FilterType): SetStatusAction => ({
+  type: 'filter/SET_STATUS',
+  payload: status,
+});
+const setClear = ():SetClearAction => ({
```

Comment:
In the action 'setClear', the purpose of this action is not clear from its name. If this action is supposed to clear the filter, rename it to 'clearFilter' to improve readability.